webclient/nginx.conf

```
@@ -14,6 +14,40 @@ http {
   access_log  /var/log/nginx/access.log  main;
   sendfile        on;
   keepalive_timeout  65;
+
+  # theme detection
+  map $cookie_theme $THEME {
+    default "light";
+
+    "dark" "dark";
+    "light" "light";
+  }
+
+  # language detection. The accept_language header sucks.
+  # This map will only be used if $cookie_lang is unset
+  map $http_accept_language $ACCEPT_LANG {
+    default "de";
+
+    "~*.*de;.*en.*" "de";
+    "~*.*en;.*de.*" "en";
+    "~*.*de,.*en.*" "de";
+    "~*.*en,.*de.*" "en";
+    "~*.*de-.*en.*" "de";
+    "~*.*en-.*de.*" "en";
+    "~*.*de;.*" "de";
+    "~*.*en;.*" "en";
+    "~*.*de,.*" "de";
+    "~*.*en,.*" "en";
+    "~*.*de.*" "de";
+    "~*.*en.*" "en";
```

```suggestion
    "~*.*de.*en.*" "de";
    "~*.*en.*de.*" "en";
    "~*.*de.*" "de";
    "~*.*en.*" "en";
```
This would be an alternative that still detects the ordering of en and de as well as thier existance. It may have false positives and edge-cases that are not handled properly.

---

How would this handle if there was no language header at all (e.g. a bot request), or if it contains neither `de` nor `en`? Can we make English the default then?

---

This is Case-agnostic byw (`"~*` is agnostic vs `"~` which is not agnostic) I cant think of a case where either one would hurt/ benefit from this beaviour. Do you?

---

> How would this handle if there was no language header at all (e.g. a bot request), or if it contains neither `de` nor `en`? Can we make English the default then?

currently `default "de";`. I dont think that we should change this, since not our whole UI (I am thinking about strings we get from the API) is translated.

---

As far as I know the standard is something like `lang-REGION`, so technically something like this could be an issue `fr-DE;en-DE`, but I doubt that this is realistic. But I'd make it case-agnostic.

---

> currently `default "de";`. I dont think that we should change this, since not our whole UI (I am thinking about strings we get from the API) is translated.

Ah, sorry, didn't see this. Yes, this is also okay for me. It's only that there is a default.